### PR TITLE
[6.5.x] DROOLS-2263 : Unexpected results in GDST when using enumerations with commas

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2263

This PR has a friend:
https://github.com/kiegroup/drools-wb/pull/767

### CODE REVIEW

**Review by manstis — APPROVED** [review]
@Rikkola Looks good to me (and nice to see a different approach to splitting the values that may potentially contain commas - my way would have been quite different!)

**Comment by manstis** [comment]
@Rikkola Presumably the fixes in your 2 PRs (for ```6.5.x```) will need replicating in ```7.5.x``` and ```master```?

**Comment by Rikkola** [comment]
@manstis Yes. Also realized I need to move this to the -api package to be able to use it both client and server side. So moving this a bit.

**Comment by Rikkola** [comment]
@manstis Ok, moved the Splitter to -api for better reusability and killed the duplicate in drools-wb

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 1]
```diff
@@ -1286,17 +1287,11 @@ protected void populateValueList(final StringBuilder buf,
                 workingValue = workingValue.substring(0,
                                                       workingValue.length() - 1);
             }
-            final String[] values = workingValue.split(",");
+
+            final String[] values = ListSplitter.split("\"",workingValue);
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java]: Just formatting, please split method arguments by one space.

**Review by jomarko — APPROVED** [review]
Nice coverage in PR. Thanks @Rikkola, just one formatting issue.

**Comment by Rikkola** [comment]
Ok. Had to add the trim variable to ListSplitter for backward+forward compatibility.

**Review by jomarko — APPROVED** [review]

**Pull request merged**